matter_server/client/client.py

```diff
@@ -251,78 +236,30 @@ async def send_command_no_wait(
             args=kwargs,
         )
-        await self._send_message(message)
+        await self.connection.send_message(message)
 
     async def connect(self) -> None:
-        """Connect to the websocket server."""
-        if self._ws_client is not None:
-            raise InvalidState("Already connected")
-
-        self.logger.debug("Trying to connect")
-        try:
-            self._ws_client = await self.aiohttp_session.ws_connect(
-                self.ws_server_url,
-                heartbeat=55,
-                compress=15,
-                max_msg_size=0,
-            )
-        except (
-            client_exceptions.WSServerHandshakeError,
-            client_exceptions.ClientError,
-        ) as err:
-            raise CannotConnect(err) from err
-
-        # at connect, the server sends a single message with the server info
-        info = cast(ServerInfoMessage, await self._receive_message_or_raise())
-        self.server_info = info
-
-        # sdk version must match exactly
-        if info.sdk_version != chip_clusters_version():
-            await self._ws_client.close()
-            raise InvalidServerVersion(
-                f"Matter Server SDK version is incompatible: {info.sdk_version} "
-                f"version on the client is {chip_clusters_version()} "
-            )
-
-        # basic check for server schema version compatibility
-        if info.schema_version < MIN_SCHEMA_VERSION:
-            # server version is too low, raise exception
-            await self._ws_client.close()
-            raise InvalidServerVersion(
-                f"Matter Server schema version is incompatible: {info.schema_version} "
-                f"a version is required that supports at least api schema {MIN_SCHEMA_VERSION} "
-                " - update the Matter Server to a more recent version."
-            )
-        if self.server_info.schema_version > MIN_SCHEMA_VERSION:
-            # server version is higher than expected, log only
-            self.logger.warning(
-                "Matter Server detected with schema version %s "
-                "which is higher than the preferred api schema %s of this client"
-                " - you may run into compatibility issues.",
-                info.schema_version,
-                MIN_SCHEMA_VERSION,
-            )
-
-        self.logger.info(
-            "Connected to Matter Fabric %s (%s), Schema version %s, CHIP SDK Version %s",
-            info.fabric_id,
-            info.compressed_fabric_id,
-            info.schema_version,
-            info.sdk_version,
-        )
+        """Connect to the Matter Server (over Websockets)."""
+        self.loop = asyncio.get_running_loop()
+        if self.connection.connected:
+            # already connected
+            return
+        await self.connection.connect()
+        # connect will raise when connecting failed,
+        # otherwise signal connected event
+        self._signal_event(EventType.CONNECTED)
```

Comment:
This is a new feature, right? I think we should wait with those.

Reply:
well, strictly takes yes but non breaking/very small. I'll take it out